plugins/lua/sort.lua

```diff
@@ -125,6 +125,109 @@ local function get_max_skill(unit_id, list)
     return max
 end
 
+local function melee_skill_effectiveness(unit_id, skill_list)
+    local unit = df.unit.find(unit_id)
```

Comment:
be sure to check that `unit` is not nil before using it. It's best not to trust that ids actually exist. (same for other added sort functions)